### ISSUE TITLE
issue: control usage with useController and useFieldArray is not correctly updated when defaultValues and values props are both present at initial render

### ISSUE THREAD

**Issue by graesj** [react-hook-form/react-hook-form#10617]
### Version Number

7.45.1

### Codesandbox/Expo snack

https://codesandbox.io/s/dazzling-joliot-4y9wk6?file=/src/App.tsx

### Steps to reproduce

1. Go to ['...'](https://codesandbox.io/s/dazzling-joliot-4y9wk6?file=/src/App.tsx)
2. Check console log 
3. Click on Submit
4. Check console log
5. See difference in form value: undefined vs true

### Expected behaviour

**Bug description:**
I discovered this potential bug when creating an UpsertComponent that either creates a new (let's say) Person object, or updates the existing Person object if it exists. In my real-world-scenario, the `values` variable is created by fetching the Person object from the backend using `React Query`. If the Person object exists, but has not yet been fetched, the form behavior is as expected (`values` take precedence over `defaultValues` once updated). However, when the Person object exists and is already cached (the API call statement returns immediately), the `defaultValues` seem to take precedence over `values` for "controlled components". In the codesandbox, you can observe the the `register("firstName")` receives the input from  `values`.

**Expected behavior:**
isLegalControlled.field.value should console.log(true) from the `values` property to `useForm`.

The same issue exists on `useFieldArray`, so I am assuming the issue is with `control`.

Is this expected behavior? If yes, how can I work around it?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
`undefined` is currently something you should try to avoid until we can remove inline defaultValue, undefined would be used to fallback to `defaultVaues`, reference: https://github.com/react-hook-form/react-hook-form/blob/master/src/useController.ts#L60-L64

**Comment by graesj** [comment]
Thank you for your fast reply (and great library)!
My initial reason for mixing `values` and `defaultValues` was due to merging functionality for CreateEntity and UpdateEntity into a single "UpsertEntity" component. After reading your reply I ended up splitting them with a shared "MutateEntityForm" component for the layout and input fields, thus avoiding mixing `values` and `defaultValues` and making the code less coupled/cluttered.

Out of curiosity, you said "**until** we can remove inline defaultValues,...".
Does this mean my initial attempt and codesandbox code will work with controlled components in the future?
